python/lsst/pipe/analysis/coaddAnalysis.py

```diff
@@ -556,27 +572,26 @@ def readCatalogs(self, patchRefList, dataset, repoInfo, fakeCat=None, raFakesCol
                 reader = afwImage.ExposureFitsReader(fname)
                 wcs = reader.readWcs()
                 mask = reader.readMask()
-                numGoodPix = len(np.where(~(mask.array.flatten() & mask.getPlaneBitMask("BAD")))[0])
+                maskBad = mask.array & 2**mask.getMaskPlaneDict()["BAD"]
+                maskNoData = mask.array & 2**mask.getMaskPlaneDict()["NO_DATA"]
+                maskedPixels = maskBad + maskNoData
+                numGoodPix = np.count_nonzero(maskedPixels == 0)
+
                 pixScale = wcs.getPixelScale().asArcseconds()
                 area = pixScale**2 * numGoodPix
                 areaDict[patchRef.dataId["patch"]] = area
 
-                patchWidth = mask.getWidth()
-                patchHeight = mask.getHeight()
-
-                patchCorners = wcs.pixelToSky([geom.Point2D(0, 0), geom.Point2D(patchWidth, patchHeight)])
+                patchCorners = wcs.pixelToSky(geom.Box2D(reader.readBBox()).getCorners())
                 areaDict["corners_" + str(patchRef.dataId["patch"])] = patchCorners
 
                 if self.config.hasFakes:
                     # Check which fake sources fall in the patch
-                    minRa = np.min(cat["coord_ra"])
-                    maxRa = np.max(cat["coord_ra"])
-                    minDec = np.min(cat["coord_dec"])
-                    maxDec = np.max(cat["coord_dec"])
-                    possOnPatch = ((fakeCat[raFakesCol].values > minRa) &
-                                   (fakeCat[raFakesCol].values < maxRa) &
-                                   (fakeCat[decFakesCol].values > minDec) &
-                                   (fakeCat[decFakesCol].values < maxDec))
+                    ras = [px.asRadians() for (px, py) in patchCorners]
+                    decs = [py.asRadians() for (px, py) in patchCorners]
+                    possOnPatch = ((fakeCat[raFakesCol].values > np.min(ras)) &
+                                   (fakeCat[raFakesCol].values < np.max(ras)) &
+                                   (fakeCat[decFakesCol].values > np.min(decs)) &
+                                   (fakeCat[decFakesCol].values < np.max(decs)))
 
```

Comment:
I think it might be clearer if these were called something like cornerRas and cornerDecs. On first reading I thought they were all the ras and decs in the catalogue.